hazelcast/src/main/java/com/hazelcast/map/impl/tx/TransactionalMapProxy.java

```diff
@@ -315,12 +319,9 @@ public void delete(Object key) {
     @SuppressWarnings("unchecked")
     public Set keySet(Predicate predicate) {
         checkTransactionState();
-        if (predicate == null) {
-            throw new NullPointerException("Predicate should not be null!");
-        }
-        if (predicate instanceof PagingPredicate) {
-            throw new NullPointerException("Paging is not supported for Transactional queries!");
-        }
+        checkNotNull(predicate, "Predicate should not be null!");
+        checkNotInstanceOf(PagingPredicate.class, predicate, "Paging is not supported for Transactional queries!");
```

Comment:
actually as @Donnerbart spotted here is a behaviour change: `checkNotInstanceOf` throws IllegalArgumentEx, but for some reason the original code is
```java
if (predicate instanceof PagingPredicate) {
            throw new NullPointerException("Paging is not supported for Transactional queries!");
}
```